ray-operator/controllers/ray/common/pod.go

```diff
@@ -793,12 +775,48 @@ func generateRayStartCommand(ctx context.Context, nodeType rayv1.RayNodeType, ra
 	return rayStartCmd
 }
 
-func addCustomAcceleratorToResourcesIfNotExists(rayStartParams map[string]string, resourceName string, resourceCount int64) error {
-	resourcesMap, err := getResourcesMap(rayStartParams)
-	if err != nil {
-		return err
+func addWellKnownAcceleratorResources(log logr.Logger, rayStartParams map[string]string, resourceLimits corev1.ResourceList) {
+	resourcesMap, _ := getResourcesMap(rayStartParams)
+
+	// Flag to track if any custom accelerator resource are present/added in rayStartParams resources.
+	isCustomAcceleratorResourceAdded := isCustomAcceleratorPresentInResources(resourcesMap)
+
+	for resourceKey, resourceValue := range resourceLimits {
+		resourceKeyString := string(resourceKey)
+
+		// Scan for resource keys ending with "gpu" like "nvidia.com/gpu"
+		if _, ok := rayStartParams["num-gpus"]; !ok {
+			if strings.HasSuffix(resourceKeyString, "gpu") && !resourceValue.IsZero() {
+				rayStartParams["num-gpus"] = strconv.FormatInt(resourceValue.Value(), 10)
+			}
+		}
+
+		// Add the first encountered custom accelerator resource from the resource limits to the rayStartParams if not already present
+		if resourcesMap != nil && !isCustomAcceleratorResourceAdded {
+			if rayResourceName, ok := customAcceleratorToRayResourceMap[resourceKeyString]; ok && !resourceValue.IsZero() {
+				if err := addCustomAcceleratorToResourcesIfNotExists(rayStartParams, resourcesMap, rayResourceName, resourceValue.Value()); err != nil {
+					log.Error(err, fmt.Sprintf("failed to add %s to resources", rayResourceName))
```

Comment:
Can we return an error instead and surface the log in generateRayStartCommand? This way you don't need to pass the logger

Reply:
Returning error, with out going through all the entries of resourceLimits, might cause missing the `num-gpus` parsing, if its later in the entries.

Reply:
If there is an error, I think we should return it and propagate it to `Reconcile` to surface the issue immediately. If we continue with the faulty CR spec, the behavior will be undefined.

Reply:
Changed the behavior to panic on any error while marshalling/unmarshalling the resources string.

Reply:
@kevin85421 what do you think? 

@andrewsykim as far as I have tested, if you specify bad resources string in the yaml, it does not get applied to the cluster and fails with error says its not a valid object.

Reply:
What's the issue with returning an error here instead of panicing though?

Reply:
@andrewsykim return `error` where? Do you mean just return error and log it, instead of panicing?

Reply:
I see where the confusion is. Typically we are following the pattern of returning and propagating errors to Reconcile to ensure errors are requeued by controller-runtime. However, I see that returning an error here would require a much larger refactor since `generateRayStartParam` and the `BuildPod` both do not return an error. 

I suggest for now we go back to logging the error instead of panicing

Reply:
Changed it back to log error, instead of panicing.